astrocut/asdf_cutouts.py

```diff
@@ -25,8 +25,17 @@ def _get_cloud_http(s3_uri: Union[str, S3Path]) -> str:
     s3_uri : string | S3Path
         the S3 URI of the cloud resource
     """
-    # create file system
-    fs = s3fs.S3FileSystem()
+    # create file system and get URL of file
+    try:
+        fs = s3fs.S3FileSystem(anon=True)
+        with fs.open(s3_uri, 'rb') as f:
+            return f.url()
+    except PermissionError:
+        # work-around for Roman Science Platform when acccessing private resources on the cloud
+        fs = s3fs.S3FileSystem()
+        with fs.open(s3_uri, 'rb') as f:
+            return f.url()
```

Comment:
Is there a way to check for permissions usage, or access, outside of try-excepting for a `PermissionError`.  I'm wondering if there's a way to consolidate this check into a conditional for `anon`, e.g.
```
        fs = s3fs.S3FileSystem(anon=is_anonymous is True)
        with fs.open(s3_uri, 'rb') as f:
            return f.url()
```
If so, that might simplify this code a bit.  Otherwise, it's fine.  In that case, would we want to issue a log.warning within the `except` block to indicate what's happening?

Reply:
Hmm, I see your point! One option would be to send an HTTP request to the URL:
```
import requests
from urllib.parse import urlparse

uri = 's3://roman-sci-test-data-prod-summer-beta-test/ROMANISIM/DENSE_REGION/R0.5_DP0.5_PA0/r0000101001001001002_01101_0001_WFI18_cal.asdf'
parsed_uri = urlparse(uri)
url = f'https://{parsed_uri.netloc}.s3.amazonaws.com/{parsed_uri.path.lstrip('/')}'

response = requests.head(url)
if response.status_code == 200:
    print("Public")
elif response.status_code == 403:
    print("Private")
else:
    print("Else")
```
I wish this could be consolidated a bit more, but it would be a more robust check then hard-coding a bucket name. I'll add this and log messages to make the process more transparent.